bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BXMLType.java

```diff
@@ -64,4 +73,26 @@ public int getTag() {
     public boolean isAnydata() {
         return true;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!super.equals(obj) || !(obj instanceof BMapType)) {
+            return false;
+        }
+
+        BXMLType other = (BXMLType) obj;
+        if (constraint == other.constraint) {
```

Comment:
Is it guaranteed that the types used as constraints have only 1 copy of that type?

Reply:
It is not. This equals is mostly for the completion purpose. These constraints are checked in jvm TypeChecker.